pkg/ocicni/ocicni.go

```diff
@@ -198,17 +202,20 @@ func initCNI(exec cniinvoke.Exec, cacheDir, defaultNetName string, confDir strin
 	if len(binDirs) == 0 {
 		binDirs = []string{DefaultBinDir}
 	}
+	// Mark whether defaultNetName is specified
+	isDefaultNetNameSpecified := defaultNetName != ""
 	plugin := &cniNetworkPlugin{
-		defaultNetName: defaultNetName,
-		networks:       make(map[string]*cniNetwork),
-		loNetwork:      getLoNetwork(exec, binDirs),
-		confDir:        confDir,
-		binDirs:        binDirs,
-		shutdownChan:   make(chan struct{}),
-		done:           &sync.WaitGroup{},
-		pods:           make(map[string]*podLock),
-		exec:           exec,
-		cacheDir:       cacheDir,
+		defaultNetName:            defaultNetName,
+		isDefaultNetNameSpecified: isDefaultNetNameSpecified,
```

Comment:
@mars1024 yeah I get that, but if you change this line to:
`isDefaultNetNameSpecified: defaultNetName != ""`
you avoid the needless allocation of a local variable

Reply:
@haircommander  DefaultNetName can be assigned through [user init](https://github.com/cri-o/ocicni/blob/b6d8b446a6448750b2341b5a1848da7dd5691428/pkg/ocicni/ocicni.go#L189) or [first-time load](https://github.com/cri-o/ocicni/blob/b6d8b446a6448750b2341b5a1848da7dd5691428/pkg/ocicni/ocicni.go#L309), if it is assigned by user init, I don't think we want it is changed no matter how cni config files change, but if it is assigned by first-time load, I think it should reload real-time.

So we can't make sure `defaultNetName != ""` is equal to `isDefaultNetNameSpecified==true`, because defaultNetName is always non-empty by user init or first-time load, and we only want to make some changes when defaultNetName is assigned by first-time load.